stock_picking_partner_note/models/stock_picking.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2024 Camptocamp (<https://www.camptocamp.com>).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+from odoo import api, fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    note = fields.Html(compute="_compute_note", store=True)
+
+    @api.depends("partner_id")
+    def _compute_note(self):
+        for picking in self:
+            pickig_type_note_type_ids = picking.mapped(
+                "picking_type_id.partner_note_type_ids"
+            )
+            partner_picking_note_ids = picking.mapped(
+                "partner_id.stock_picking_note_ids"
+            )
+            picking_notes = []
```

Comment:
Maybe enough like that:
```suggestion
            picking_notes = picking.partner_id.stock_picking_note_ids \
            .filtered(lambda n: n.note_type_id in picking_type_note_type_ids) \
            .mapped("name")

```